src/index.js

```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-exports.DAGNode = require('./dag-node.js')
+exports.DAGNode = require('./dag-node-factory.js')
```

Comment:
We can't expose the DAGNode class. If we do that, DAGNode will become *mutable*, ie. doing `const node = new DAGNode(...)` becomes possible which is not what we want.

The general idea of the factory pattern is to expose only methods for creating the DAGNodes, not for users to directly use DAGNode as it was before. This is the first step to immutability.